internal/bminventory/inventory.go

```diff
@@ -2289,6 +2296,15 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 		return installer.NewRegisterHostInternalServerError().
 			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
 	}
+
+	if b.crdUtils != nil && cluster.KubeKeyNamespace != "" {
```

Comment:
Maybe instead of checking for `b.crdUtils` we should have a dummy `CRDUtils`?
It will make the code cleaner no?

Reply:
Sure, good idea

Reply:
Done